src/dbnode/persist/fs/types.go

```diff
@@ -145,18 +142,28 @@ type DataFileSetReader interface {
 	// Status returns the status of the reader
 	Status() DataFileSetReaderStatus
 
-	// Read returns the next id, tags, data, checksum tuple or error, will return io.EOF at end of volume.
-	// Use either Read or ReadMetadata to progress through a volume, but not both.
-	// Note: make sure to finalize the ID, close the Tags and finalize the Data when done with
-	// them so they can be returned to their respective pools.
-	Read() (id ident.ID, tags ident.TagIterator, data checked.Bytes, checksum uint32, err error)
-
 	// StreamingRead returns the next unpooled id, encodedTags, data, checksum
 	// values ordered by id, or error, will return io.EOF at end of volume.
 	// Can only by used when DataReaderOpenOptions.StreamingEnabled is true.
 	// Note: the returned data gets invalidated on the next call to StreamingRead.
 	StreamingRead() (StreamedDataEntry, error)
 
+	// StreamingReadMetadata returns the next unpooled id, encodedTags, length checksum
+	// values ordered by id, or error; will return io.EOF at end of volume.
+	// Can only by used when DataReaderOpenOptions.StreamingEnabled is true.
+	// Use either StreamingRead or StreamingReadMetadata to progress through a volume, but not both.
+	// Note: the returned id and encodedTags get invalidated on the next
+	// call to StreamingReadMetadata.
+	StreamingReadMetadata() (
```

Comment:
nit; `StreamingMetadata` perhaps?

Reply:
I'd prefer to stick with the current naming as it gives a consistent progression:
`Read` -> `StreamingRead`,
`ReadMetadata` -> `StreamingReadMetadata`.